sentry/src/main/java/io/sentry/SentryEnvelopeItem.java

```diff
@@ -86,24 +64,24 @@
     }
   }
 
-  public static @NotNull SentryEnvelopeItem fromEvent(
-      final @NotNull ISerializer serializer, final @NotNull SentryEvent event) throws IOException {
-    Objects.requireNonNull(serializer, "ISerializer is required.");
-    Objects.requireNonNull(event, "SentryEvent is required.");
+  public static @NotNull SentryEnvelopeItem from(
+      final @NotNull ISerializer serializer, final @NotNull ConvertibleToEnvelopeItem item)
```

Comment:
Can we get away from having `ConvertibleToEnvelopeItem` altogether?

Seems it only exists as a marker and for `sentryItemType`. 
Would it be possible to take `Object` here (or `<T>` since it's what `serialize` takes) and add types for the known types (SentryEvent, Session, SentryTransaction) and simply leave the default as `attachment` or just not type at all if that's possible?

The trade off is not having the compile time check that an envelope item must be a type that returns a String here, vs not having to implement this interface on any type we want to create as an envelope item.

Reply:
it could be `<T extends SentryItem>` I guess as well

Reply:
Can you pass any object as a `SentryEnvelopeItem`? If so, we can remove it and make this method accept just `Object` but if not, I think that checking with `instance of` and throwing exception (or even worse failing silently) if object is not one of the supported ones is more confusing and error prone.

Reply:
Yes, we'll have attachments soon (this quarter) and that'll be just a data blob or stream.